docs/sphinx/dev/dev_style_guide_code.rst

```diff
@@ -1,28 +1,169 @@
+Avoid `using namespace X;`. Instead specify where you are using classes or
+functions from a namespace, e.g. `std::vector<std::string> my variable;`.
+Never do `using namespace std;`.
 
-Explicit typing should be used wherever possible, e.g. always use ``uint32_t`` rather than ``uint``.
+Header files
+^^^^^^^^^^^^^
+C++ header files should be named after the class they describe and use the
+ending `.hpp`. They should begin with a header file commit like
 
+.. code-block:: c++
 
-Variables
-^^^^^^^^^^
+    /****************************************************
+    @file
+    @brief Brief description of the classes in this file.
+    - MyClass
+    - MyOtherClass : public MyOtherBaseClass
+    *****************************************************/
 
-Structs
-^^^^^^^^^^
+Immediately after the header file comment include guards should follow:
 
-Enums
-^^^^^^^^^^
+.. code-block:: c++
+
+    #ifndef MY_CLASS_H
+    #define MY_CLASS_H
+
+With a simple
+
+.. code-block:: c++
+
+    #endif
+
+at the end of the file.
+
+
+#include Style
+^^^^^^^^^^^^^^^
+The list of #includes should be in the order:
+
+ - Header file implementing the interface of this `.cpp` file.
+ - kotekan project headers
+ - Local/Private Headers
+ - System #includes
+
+Each category should be sorted lexicographically by the full path.
+
+The small details
+^^^^^^^^^^^^^^^^^^
+
+- Access modifiers (like `public`, `private`, etc.) should have indentation level
+  `0`.
+
+- Don't indent namespaces.
+
+- Pointer and reference symbols should be left aligned, e.g.
+
+  .. code-block:: c++
+
+      int* a;
+      int& b;
+
+- Operands of binary and ternary expressions, trailing comments should be
+  horizontally aligned if
+  they are split over multiple lines, e.g.:
+
+  .. code-block:: c++
+
+      int a = abc *     // My comment
+              def;      // is quite long.
+      bool aaaaa = bbbbbbbbbbbbbbbbbbbb
+                   && ccccccccccccccccccccc;
+
+- The parameters in a function definition or declaration as well as the
+  arguments in a function call should either be all in one line or in one line
+  each and horizontally aligned, e.g.:
+
+  .. code-block:: c++
+
+      int a = f(b, c, d, e);
+      int f(int b, int c, int d, int e);
+      int f(int b, int c, int d, int e) {}
+
+  .. code-block:: c++
+
+      int a = f(
+          bbb, ccc, ddd, eee);
+      int f(
+          int bbb, int ccc, int ddd, int edd);
+      int f(
+          int bbb, int ccc, int ddd, int eee) {}
+
+  .. code-block:: c++
+
+      int a = f(bbbbbb,
+                cccccc,
+                dddddd,
+                eeeeee);
+      int f(int bbbbbb,
+            int cccccc,
+            int dddddd,
+            int eeeeee);
+      int f(int bbbbbb,
+            int cccccc,
+            int dddddd,
+            int eeeeee) {}
+
+- Don't add a newline before an opening curly bracket, e.g.:
+
+  .. code-block:: c++
+
+      void f(bool a) {
```

Comment:
I've had mixed feelings about this one. I quite like Strousop style placing the `{` on the next line (for functions/methods only).

Reply:
I know it looks like Java, but like Rick, I also like keeping the opening brace on the same line as the function signature.

Reply:
Not *just* Java, Go also keeps the opening function brace on the same line. :-)